### ISSUE TITLE
wireguird under wayland

### ISSUE THREAD

**Issue by cavaughankirov** [UnnoTed/wireguird#54]
Was trying out wireguird under Cosmic  DE and it will not open. I think the reason has to do with the fact that wayland will not allow apps running as root (sudo?) to run. Regardless of the reason would love to see this resolved..

**Comment by mimi0000oo** [comment]
It does not work under hyprland. I think something has to do with wayland/sudo. I've seen some solutions online for other programs with this kind of programs with using something called [ego](https://github.com/intgr/ego), but it does not seem to work with wireguird

**Comment by mimi0000oo** [comment]
after a bit more research i found this script. maybe it helps you.
```
#!/bin/bash
#small script to enable root access to x-windows system 
xhost +SI:localuser:root
sudo $1
#disable root access after application terminates
xhost -SI:localuser:root
#print access status to allow verification that root access was removed
xhost
```
source: https://www.reddit.com/r/Fedora/comments/5eb633/solution_running_graphical_app_with_sudo_in/

**Comment by shayaknyc** [comment]
> after a bit more research i found this script. maybe it helps you.
> 
> ```
> #!/bin/bash
> #small script to enable root access to x-windows system 
> xhost +SI:localuser:root
> sudo $1
> #disable root access after application terminates
> xhost -SI:localuser:root
> #print access status to allow verification that root access was removed
> xhost
> ```
> 
> source: https://www.reddit.com/r/Fedora/comments/5eb633/solution_running_graphical_app_with_sudo_in/

Created this script, and used it to execute `wsudo wireguird` from the terminal, and it loaded. That said, it runs in the context of the `root` user, and when I activated the tunnel, it seemed to connect without issue, but didn't seem to actually work on my user account. So, not sure this ultimately worked..... Unless, I'm missing something?

**Comment by mimi0000oo** [comment]
> So, not sure this ultimately worked..... Unless, I'm missing something?

I don't think you are missing anything. if the gui opens up and you can make a valid connection, then it worked. Now, all the traffic makes it through wireguard.

The problem that this solves is that it opens the gui as root in wayland. It is needed as wayland does not let you run anything under sudo, as far as i remember.
